File: Level 1/Defibrillators.dart
import "dart:io" show stdin;
import "dart:math" show PI, cos, sqrt;

void main() {
  var longitude = readRadians();
  var latitude = readRadians();

  var n = readInt();
  var defibrillators = readDefibrillators(n);

  var solution;
  var min;

  for (var defibrillator in defibrillators) {

    var x = (longitude - defibrillator.longitude)
        * cos((defibrillator.latitude + latitude) / 2);
    var y = latitude - defibrillator.latitude;
    var d = sqrt(x * x + y * y) * 6371;

    if (min == null || d < min) {
      min = d;

      solution = defibrillator.name;
    }
  }

  print(solution);
}

class Defibrillator {
  final String name;
  final double longitude;
  final double latitude;

  const Defibrillator(this.name, this.longitude, this.latitude);
}

String readString() => stdin.readLineSync();

int readInt() => int.parse(readString());

double readRadians() => toRadians(readString());

List<Defibrillator> readDefibrillators(int n)
  => new List<Defibrillator>.generate(n, (_) => readDefibrillator());

Defibrillator readDefibrillator() {
  var line = readString().split(";");

  return new Defibrillator(line[1], toRadians(line[4]), toRadians(line[5]));
}

double toRadians(String value)
  => double.parse(value.replaceFirst(",", ".")) * (PI / 180.0);

File: Level 1/Defibrillators.scala
// Defibrillators
// Read inputs from System.in, Write outputs to use print.
// Your class name has to be Solution
import math._
import scala.util._

object Solution {
  def main(args: Array[String]) {
var longitude = readLine.replaceAll(",",".").toFloat
var latitude = readLine.replaceAll(",",".").toFloat
var n = readInt
var addresses = Array[String]();
for (i <- 0 to n - 1) {
    addresses = addresses :+ readLine
}

var minDistance = 10000000.toFloat;
var minName = "";
for (i <- 0 to n - 1) {
    var distance = calcDistance(longitude, addresses(i).split(";")(4).replaceAll(",",".").toFloat,
      latitude, addresses(i).split(";")(5).replaceAll(",",".").toFloat); 
    if (distance < minDistance) {
        minDistance = distance;
        minName = addresses(i).split(";")(1);
    }
}
println(minName);


  }
def calcDistance(longA:Float, longB:Float, latiA:Float, latiB:Float): Float = {
    var x = (longB - longA) * Math.cos((latiA+latiB)/2);
    var y = latiB - latiA;
    var result = sqrt(x*x + y*y) * 6371;
    return result.toFloat
}

}
